main_test.go

```diff
@@ -12,7 +12,7 @@ func TestAesEncrypt(t *testing.T) {
 
 	key, err := randBytes(aesKeyLength)
 	require.NoError(t, err)
-	ciphertext, tag, iv, err := AesEncrypt(key, data)
+	ciphertext, tag, iv, err := AesEncrypt(key, data, make([]byte, 0))
 	require.NoError(t, err)
 
```

Comment:
add test for self-provided `iv`.  assert returned `iv` be what you provided. expect decryption to work as expected.

Reply:
test giving invalid `iv` expect error.

Reply:
add test expecting deterministic encryption without random `iv`